core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java

```diff
@@ -167,8 +168,16 @@ private ArtifactSignerProvider loadSigners(
             "yaml",
             new YamlSignerParser(List.of(artifactSignerFactory))));
 
-    if (config.getAzureKeyVaultParameters() != null) {
+    if (config.getAzureKeyVaultParameters().isAzureKeyVaultEnabled()) {
       final AzureKeyVaultParameters params = config.getAzureKeyVaultParameters();
+      if (!ObjectUtils.allNotNull(
+          params.getClientlId(),
+          params.getClientSecret(),
+          params.getTenantId(),
+          params.getKeyVaultName())) {
+        throw new InitializationException(
```

Comment:
Will this show up as a nice error message on the cli? Not just a java exception with stacktrace?

Reply:
yeah, had to do a little refactor to make this come out nicely.